plugins/gatsby-source-wagtail/gatsby-node.js

```diff
@@ -0,0 +1,38 @@
+const fetch = require('node-fetch');
+
+exports.sourceNodes = async (
+  { actions, createNodeId, createContentDigest },
+  configOptions,
+) => {
+  const { createNode } = actions;
+  // Gatsby adds a configOption that's not needed for this plugin, delete it
+  delete configOptions.plugins; // eslint-disable-line
+  const fetchBrandingData = async () => {
+    try {
+      const response = await fetch(configOptions.pagesApiUrl);
+      return await response.json();
+    } catch (e) {
+      return console.error(`${e.name}: ${e.message}`); // eslint-disable-line
+    }
+  };
+
+  // the gatsby node shape will be the exact same as what the api returns.
+  const processNode = (node, nodeType) => createNode({
+    ...node,
+    id: createNodeId(`wagtail-node-${node.id}`),
+    parent: null,
+    children: [],
```

Comment:
In the docs for `createNode` they were not marked as optional, but it seems that they aren't needed after all. I removed them since we don't use them.